glib/src/win32.rs

```diff
@@ -1,16 +1,103 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
-use std::path::PathBuf;
+use std::{ffi::OsString, path::PathBuf};
 
 use crate::translate::*;
 
-#[doc(alias = "g_win32_get_package_installation_directory_of_module")]
-pub fn win32_get_package_installation_directory_of_module(
-    hmodule: ffi::gpointer,
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub enum OSType {
+    #[doc(alias = "G_WIN32_OS_ANY")]
+    Any,
+    #[doc(alias = "G_WIN32_OS_WORKSTATION")]
+    Workstation,
+    #[doc(alias = "G_WIN32_OS_SERVER")]
+    Server,
+}
+
+#[doc(hidden)]
+impl IntoGlib for OSType {
+    type GlibType = ffi::GWin32OSType;
+
+    #[inline]
+    fn into_glib(self) -> Self::GlibType {
+        match self {
+            Self::Any => ffi::G_WIN32_OS_ANY,
+            Self::Workstation => ffi::G_WIN32_OS_WORKSTATION,
+            Self::Server => ffi::G_WIN32_OS_SERVER,
+        }
+    }
+}
+
+#[doc(alias = "g_win32_check_windows_version")]
+pub fn check_windows_version(major: i32, minor: i32, spver: i32, os_type: OSType) -> bool {
+    unsafe {
+        from_glib(ffi::g_win32_check_windows_version(
+            major,
+            minor,
+            spver,
+            os_type.into_glib(),
+        ))
+    }
+}
+
+#[doc(alias = "g_win32_get_command_line")]
+#[doc(alias = "get_command_line")]
+pub fn command_line() -> Vec<OsString> {
+    unsafe { FromGlibPtrContainer::from_glib_full(ffi::g_win32_get_command_line()) }
+}
+
+#[doc(alias = "g_win32_error_message")]
+pub fn error_message(error: i32) -> OsString {
+    unsafe { from_glib_full(ffi::g_win32_error_message(error)) }
+}
+
+#[doc(alias = "g_win32_getlocale")]
+pub fn getlocale() -> OsString {
+    unsafe { from_glib_full(ffi::g_win32_getlocale()) }
+}
+
+#[doc(alias = "g_win32_get_package_installation_directory")]
+#[doc(alias = "get_package_installation_directory")]
+pub fn package_installation_directory(
+    package: Option<&str>,
+    dll_name: Option<&str>,
 ) -> Option<PathBuf> {
+    unsafe {
+        from_glib_full(ffi::g_win32_get_package_installation_directory(
+            package.to_glib_none().0,
+            dll_name.to_glib_none().0,
+        ))
+    }
+}
+
+#[doc(alias = "g_win32_get_package_installation_directory_of_module")]
+#[doc(alias = "get_package_installation_directory_of_module")]
+pub fn package_installation_directory_of_module(hmodule: ffi::gpointer) -> Option<PathBuf> {
     unsafe {
         from_glib_full(ffi::g_win32_get_package_installation_directory_of_module(
             hmodule,
         ))
     }
 }
+
+#[doc(alias = "g_win32_get_package_installation_subdirectory")]
+#[doc(alias = "get_package_installation_subdirectory")]
+pub fn package_installation_subdirectory(
+    package: Option<&str>,
+    dll_name: Option<&str>,
+    subdir: Option<&str>,
+) -> Option<PathBuf> {
+    unsafe {
+        from_glib_full(ffi::g_win32_get_package_installation_subdirectory(
```

Comment:
> g_win32_get_package_installation_subdirectory has been deprecated since version 2.18 and should not be used in newly-written code.
Pass the HMODULE of a DLL or EXE to [g_win32_get_package_installation_directory_of_module()](https://github.com/gtk-rs/gtk-rs-core/pull/1042/glib-Windows-Compatibility-Functions.html#g-win32-get-package-installation-directory-of-module) instead, and then construct a subdirectory pathname with [g_build_filename()](https://github.com/gtk-rs/gtk-rs-core/pull/1042/glib-Miscellaneous-Utility-Functions.html#g-build-filename).

So let's maybe not add this

Reply:
Should `g_win32_get_windows_version` be removed as well? It's also been deprecated since GLib 2.44.

Reply:
Yes, let's not add deprecated things :)